legal-api/tests/unit/__init__.py

```diff
@@ -38,12 +40,18 @@ def has_expected_date_str_format(date_str: str, format: str) -> bool:
 
 
 @contextmanager
-def nested_session(session, exception=Exception):
+def nested_session(session):
     try:
         sess = session.begin_nested()
         yield sess
         sess.rollback()
-    except exception:
+    except AssertionError as err:
+        raise err
+    except exc.ResourceClosedError as err:
+        # mean the close out of the transaction got fouled in pytest
+        err
```

Comment:
added this to avoid "F841 local variable 'err' is assigned to but never used." Please let me know if a better way to handle this.

Reply:
does it actually complain if you just use the following?

```python
except exc.ResourceClosedError:
        # mean the close out of the transaction got fouled in pytest
```

Reply:
> does it actually complain if you just use the following?
> 
> ```python
> except exc.ResourceClosedError:
>    # mean the close out of the transaction got fouled in pytest
>   pass
> ```

Does not  complain (after removing `err`). Seems to be working this way